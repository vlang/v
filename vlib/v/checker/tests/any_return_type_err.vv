struct Test {
}

fn (f Test) abc() any {
	return 0
}

fn (f Test) abc2() ?any {
	return 0
}

fn (f Test) abc3() !any {
	return 0
}

fn foo() any {
	return 3
}

fn foo2() ?any {
	return 3
}

fn foo3() !any {
	return 3
}

fn main() {
	println('any ${foo}')
}
