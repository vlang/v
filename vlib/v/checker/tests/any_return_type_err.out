vlib/v/checker/tests/any_return_type_err.vv:4:19: error: cannot use type `any` here
    2 | }
    3 | 
    4 | fn (f Test) abc() any {
      |                   ~~~
    5 |     return 0
    6 | }
vlib/v/checker/tests/any_return_type_err.vv:8:20: error: cannot use type `any` here
    6 | }
    7 | 
    8 | fn (f Test) abc2() ?any {
      |                    ~~~~
    9 |     return 0
   10 | }
vlib/v/checker/tests/any_return_type_err.vv:12:20: error: cannot use type `any` here
   10 | }
   11 | 
   12 | fn (f Test) abc3() !any {
      |                    ~~~~
   13 |     return 0
   14 | }
vlib/v/checker/tests/any_return_type_err.vv:16:10: error: cannot use type `any` here
   14 | }
   15 | 
   16 | fn foo() any {
      |          ~~~
   17 |     return 3
   18 | }
vlib/v/checker/tests/any_return_type_err.vv:20:11: error: cannot use type `any` here
   18 | }
   19 | 
   20 | fn foo2() ?any {
      |           ~~~~
   21 |     return 3
   22 | }
vlib/v/checker/tests/any_return_type_err.vv:24:11: error: cannot use type `any` here
   22 | }
   23 | 
   24 | fn foo3() !any {
      |           ~~~~
   25 |     return 3
   26 | }
